internal/status/cache.go

```diff
@@ -53,8 +54,9 @@ type CacheEntry interface {
 type Cache struct {
 	proxyUpdates map[types.NamespacedName]*ProxyUpdate
 
-	gatewayRef   types.NamespacedName
-	routeUpdates map[types.NamespacedName]*ConditionsUpdate
+	gatewayRef     types.NamespacedName
+	gatewayUpdates map[types.NamespacedName]*GatewayConditionsUpdate
+	routeUpdates   map[types.NamespacedName]*RouteConditionsUpdate
```

Comment:
I'd rename this back to `ConditionsUpdate` and name the var the same. Originally it was HTTPRoute, then we added TLSRoute, so it just became `Route`, but could change to plain conditions since that's what its processing.

Reply:
@stevesloka we need to have separate types for updating route and gateway conditions. With that said, keeping the `ConditionsUpdate` name makes it difficult to differentiate between a gateway and a route condition update.